Code/Source/solver/nn.cpp

```diff
@@ -540,6 +541,104 @@ void gnn(const int eNoN, const int nsd, const int insd, Array<double>& Nxi, Arra
   }
 }
 
+/// @brief This routine returns a surface normal vector at element "e" and Gauss point
+/// 'g' of a virtual face 'lFa' weighted by Jac, i.e.
+/// Jac = SQRT(NORM(n)), the Jacobian of the mapping from parent surface element to
+/// reference/old/new configuration.
+///
+/// Since virtual faces do not have an interior element, the direction of the normal vector
+/// is assumed from the nodal ordering.
+///
+/// cfg denotes which configuration (reference/timestep 0, old/timestep n, or new/timestep n+1).
+///
+/// Reproduces Fortran 'GNNBv'.
+//
+void gnnbv(const ComMod& com_mod, const CmMod& cm_mod, const faceType& lFa, const int e, const int g, const int nsd, const int insd,
+  const int eNoNb, const Array<double>& Nx, Vector<double>& n, MechanicalConfigurationType cfg)
+{
```

Comment:
This is no longer a Fortran code so we should use more meaningful variable names

cfg -> mech_config
n -> surface_normal